backend/corpora/common/utils/db_utils.py

```diff
@@ -29,3 +31,25 @@ def query(self, table_args: typing.List[Base], filter_args: typing.List[bool] =
             if filter_args
             else self.session.query(*table_args).all()
         )
+
+    retry_limit = 3  # The number of times to attempt generating a uuid
+
+    def generate_id(self, table: Base, *args):
+        """
+        Generates a UUID to enter into the specified table.
+
+        :param table: The table to generate the uuid  for
+        :param arg: Additional primary keys if needed
+        :return:
+        """
+        # Generate the ID
+        retry_attempts = 0
+        while retry_attempts < self.retry_limit:
+            _id = str(uuid.uuid4())
+            key = _id if not args else (_id, *args)
+            if self.get(table, key):
```

Comment:
A collision is rather unlikely, no?

Reply:
The chance is low, but the cost of check is also low. The best case is it never happens. The worst case is we silently overwrite someones published project. I'm trying to mitigate the worst case.

Reply:
>The chance is low

"Low" kind of understates it, see https://en.wikipedia.org/wiki/Universally_unique_identifier#Collisions